rqt_plot/src/rqt_plot/data_plot/__init__.py

```diff
@@ -210,12 +211,37 @@ def save_settings(self, plugin_settings, instance_settings):
         Currently, this is just the plot type, but may include more useful
         data in the future"""
         instance_settings.set_value('plot_type', self._plot_index)
+        xlim = self.get_xlim()
+        ylim = self.get_ylim()
+        # convert limits to normal arrays of floats; some backends return numpy
+        # arrays
+        xlim = [ float(x) for x in xlim ]
+        ylim = [ float(y) for y in ylim ]
+        instance_settings.set_value('x_limits', pack(xlim))
+        instance_settings.set_value('y_limits', pack(ylim))
 
     def restore_settings(self, plugin_settings, instance_settings):
         """Restore the settings for this widget
 
         Currently, this just restores the plot type."""
         self._switch_data_plot_widget(int(instance_settings.value('plot_type', 0)))
+        xlim = unpack(instance_settings.value('x_limits', []))
+        ylim = unpack(instance_settings.value('y_limits', []))
+        if xlim:
+            # convert limits to an array of floats; they're often lists of
+            # strings
+            try:
+                xlim = [ float(x) for x in xlim ]
+                self.set_xlim(xlim)
+            except:
```

Comment:
The set_xlim call might throw a number of different exceptions depending on which backend is in use.

I'm now aware of any exception which shouldn't be caught here, and the fall-back behavior is reasonable in all cases.